pkg/operation/botanist/controlplane/kubecontrollermanager/kube_controller_manager.go

```diff
@@ -0,0 +1,529 @@
+	}
+
+	if k.config.HorizontalPodAutoscalerConfig != nil {
+		if k.config.HorizontalPodAutoscalerConfig.CPUInitializationPeriod != nil {
+			horizontalPodAutoscalerConfig.CPUInitializationPeriod = k.config.HorizontalPodAutoscalerConfig.CPUInitializationPeriod
+		}
+		if k.config.HorizontalPodAutoscalerConfig.DownscaleDelay != nil {
+			horizontalPodAutoscalerConfig.DownscaleDelay = k.config.HorizontalPodAutoscalerConfig.DownscaleDelay
+		}
+		if k.config.HorizontalPodAutoscalerConfig.DownscaleStabilization != nil {
+			horizontalPodAutoscalerConfig.DownscaleStabilization = k.config.HorizontalPodAutoscalerConfig.DownscaleStabilization
+		}
+		if k.config.HorizontalPodAutoscalerConfig.InitialReadinessDelay != nil {
+			horizontalPodAutoscalerConfig.InitialReadinessDelay = k.config.HorizontalPodAutoscalerConfig.InitialReadinessDelay
+		}
+		if k.config.HorizontalPodAutoscalerConfig.SyncPeriod != nil {
+			horizontalPodAutoscalerConfig.SyncPeriod = k.config.HorizontalPodAutoscalerConfig.SyncPeriod
+		}
+		if k.config.HorizontalPodAutoscalerConfig.Tolerance != nil {
+			horizontalPodAutoscalerConfig.Tolerance = k.config.HorizontalPodAutoscalerConfig.Tolerance
+		}
+		if k.config.HorizontalPodAutoscalerConfig.UpscaleDelay != nil {
+			horizontalPodAutoscalerConfig.UpscaleDelay = k.config.HorizontalPodAutoscalerConfig.UpscaleDelay
+		}
+	}
+	return horizontalPodAutoscalerConfig
+}
+
+var (
+	versionConstraintK8sEqual113        *semver.Constraints
+	versionConstraintK8sGreaterEqual112 *semver.Constraints
+	versionConstraintK8sSmaller112      *semver.Constraints
+	versionConstraintK8sGreaterEqual113 *semver.Constraints
+	versionConstraintK8sSmaller114      *semver.Constraints
+	versionConstraintK8sGreaterEqual115 *semver.Constraints
+	versionConstraintK8sGreaterEqual116 *semver.Constraints
+	versionConstraintK8sGreaterEqual117 *semver.Constraints
+)
+
+func init() {
+	var err error
+
+	versionConstraintK8sEqual113, err = semver.NewConstraint("~ 1.13")
+	utilruntime.Must(err)
+	versionConstraintK8sGreaterEqual112, err = semver.NewConstraint(">= 1.12")
+	utilruntime.Must(err)
+	versionConstraintK8sSmaller112, err = semver.NewConstraint("< 1.12")
+	utilruntime.Must(err)
+	versionConstraintK8sSmaller114, err = semver.NewConstraint("< 1.14")
+	utilruntime.Must(err)
+	versionConstraintK8sGreaterEqual113, err = semver.NewConstraint(">= 1.13")
+	utilruntime.Must(err)
+	versionConstraintK8sGreaterEqual115, err = semver.NewConstraint(">= 1.15")
+	utilruntime.Must(err)
+	versionConstraintK8sGreaterEqual116, err = semver.NewConstraint(">= 1.16")
+	utilruntime.Must(err)
+	versionConstraintK8sGreaterEqual117, err = semver.NewConstraint(">= 1.17")
+	utilruntime.Must(err)
+}
+
+// Secrets is collection of secrets for the kube-controller-manager.
+type Secrets struct {
+	// Kubeconfig is a secret that contains a kubeconfig which can be used by the kube-controller-manager to communicate with the kube-apiserver.
+	Kubeconfig Secret
```

Comment:
```suggestion
	Kubeconfig component.Secret
```

(same for other attributes)